.circleci/config.yml

```diff
@@ -97,6 +97,77 @@ commands:
+            cf api "$CF_ENDPOINT"
+            cf auth "$CF_USER" "$CF_PASSWORD"
+            cf add-plugin-repo CF-Community https://plugins.cloudfoundry.org
+            cf install-plugin app-autoscaler-plugin -r CF-Community -f
+            cf target -o "$CF_ORG" -s development
+      - run:
+          name: "Fetch existing manifest"
+          command: |
+            cf create-app-manifest "$CF_APP-dev" -p deploy_manifest.yml
+      - run:
+          name: "Update app manifest"
+          command: |
+            sed -i  "s/instances: [0-9]\+/instances\: 1/" deploy_manifest.yml
+      - run:
+          name: "Push review app"
+          command: |
+            export DOCKER_IMAGE=tariff-frontend
+            export DOCKER_TAG=dev-${CIRCLE_SHA1}
+            export PR_NUMBER=$(echo $CIRCLE_PULL_REQUEST | sed 's|https://github.com/||' | cut -d '/' -f 4)
+            export REVIEW_URL_ID="pr${PR_NUMBER}"
+
+            CF_DOCKER_PASSWORD=$AWS_SECRET_ACCESS_KEY cf push "$CF_APP-${REVIEW_URL_ID}" -f deploy_manifest.yml --no-route --docker-image "$ECR_REPO/$DOCKER_IMAGE:$DOCKER_TAG" --docker-username "$AWS_ACCESS_KEY_ID"
+            cf map-route "$CF_APP-$REVIEW_URL_ID" london.cloudapps.digital -n "$CF_APP-$REVIEW_URL_ID"
+
+            # Enable routing from this service to the backend applications which are private
+            cf add-network-policy "$CF_APP-$REVIEW_URL_ID" "$CF_BACKEND_APP_XI-dev" --protocol tcp --port 8080
+            cf add-network-policy "$CF_APP-$REVIEW_URL_ID" "$CF_BACKEND_APP_UK-dev" --protocol tcp --port 8080
+
+  cf_purge_reviews:
+    steps:
+      - run:
+          name: "Setup CF CLI"
+          command: |
+            curl -L -o cf.deb --retry 3 'https://packages.cloudfoundry.org/stable?release=debian64&version=7.2.0&source=github-rel'
+            file cf.deb
+            sudo dpkg -i cf.deb
+            cf -v
+            cf api "$CF_ENDPOINT"
+            cf auth "$CF_USER" "$CF_PASSWORD"
+            cf add-plugin-repo CF-Community https://plugins.cloudfoundry.org
+            cf install-plugin app-autoscaler-plugin -r CF-Community -f
+            cf target -o "$CF_ORG" -s development
+      - run:
+          name: Find PRs
+          command: |
+            REPO_API_URL=$(echo "${CIRCLE_REPOSITORY_URL}/" | sed 's|.git/$||' | sed 's|git@github.com:|https://api.github.com/repos/|')
+            curl "${REPO_API_URL}/pulls?state=open" | jq '.[].number' > open_prs.txt
+            cat open_prs.txt | sed "s/^/${CF_APP}-pr/" | sort > expected_apps.txt
+            cat expected_apps.txt
+      - run:
+          name: Find old review apps
+          command: |
+            cf apps | cut -d ' ' -f 1  | grep "${CF_APP}-pr" | sort > current_apps.txt
+            (comm -1 --output-delimiter="|" <(cat expected_apps.txt) <(cat current_apps.txt) | egrep -v '^\|' || true) > orphaned_apps.txt
+      - run:
+          name: Remove orphaned apps
+          command: |
+            cat orphaned_apps.txt | while read ORPHANED_APP
```

Comment:
```suggestion
            for ORPHANED_APP in $(cat orphaned_apps.txt)
```

Reply:
Opinionated nonsense but I do find ^^ a bit more readable

Reply:
Yeh - totally agree - I'll get that changed - (some my shell scripting definitely gives my age away :laughing: )